test/common/broker.go

```diff
@@ -1,73 +1,246 @@
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		auth := req.Header.Get("Authorization")
+		if auth == "" {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("Missing authorization header"))
+			return
 		}
-		var reqData interface{}
-		err = json.Unmarshal(b.RawRequestBody, &reqData)
+		const basicHeaderPrefixLength = len("Basic ")
+		decoded, err := base64.StdEncoding.DecodeString(auth[basicHeaderPrefixLength:])
 		if err != nil {
-			panic(err)
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		if string(decoded) != fmt.Sprintf("%s:%s", b.Username, b.Password) {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("Credentials mismatch"))
+			return
 		}
-		b.RequestBody = httpexpect.NewValue(GinkgoT(), reqData)
+		next.ServeHTTP(w, req)
+	})
+}
 
-	case http.MethodGet:
-		if responseBody == nil && req.URL.Path == "/v2/catalog" {
-			responseBody = []byte(serviceCatalog)
+func (b *BrokerServer) saveRequestMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		b.LastRequest = req
+		bodyBytes, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			SetResponse(w, http.StatusInternalServerError, Object{
+				"description": "Could not read body",
+				"error":       err.Error(),
+			})
+			return
 		}
+		b.LastRequestBody = bodyBytes
+		next.ServeHTTP(w, req)
+	})
+}
+
+func (b *BrokerServer) defaultCatalogHandler(rw http.ResponseWriter, req *http.Request) {
+	SetResponse(rw, http.StatusOK, b.Catalog)
+}
+
+func (b *BrokerServer) defaultServiceInstanceHandler(rw http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodPut {
+		SetResponse(rw, http.StatusCreated, Object{})
+	} else {
+		SetResponse(rw, http.StatusOK, Object{})
 	}
+}
 
-	code := b.StatusCode
-	if code == 0 {
-		code = http.StatusOK
+func (b *BrokerServer) defaultServiceInstanceLastOpHandler(rw http.ResponseWriter, req *http.Request) {
+	SetResponse(rw, http.StatusOK, Object{
+		"state": "succeeded",
+	})
+}
+
+func (b *BrokerServer) defaultBindingHandler(rw http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodPut {
+		SetResponse(rw, http.StatusCreated, Object{
+			"credentials": Object{
+				"instance_id": mux.Vars(req)["instance_id"],
+				"binding_id":  mux.Vars(req)["binding_id"],
+			},
+		})
+	} else {
+		SetResponse(rw, http.StatusOK, Object{})
 	}
-	rw.Header().Set("Content-Type", "application/json")
-	rw.WriteHeader(code)
+}
 
-	rw.Write(responseBody)
+func (b *BrokerServer) defaultBindingLastOpHandler(rw http.ResponseWriter, req *http.Request) {
+	SetResponse(rw, http.StatusOK, Object{
+		"state": "succeeded",
+	})
 }
 
-func (b *Broker) Called() bool {
-	return b.Request != nil
+func SetResponse(rw http.ResponseWriter, status int, message interface{}) {
+	rw.Header().Set("Content-Type", "application/json")
```

Comment:
you could remove this and in the if else reuse the pkg/util.go/WriteJSON method - also I am not sure if in general we need the extra test util method SetResponse